client/xdag_config.c

```diff
@@ -13,111 +11,93 @@
 #define false (0)
 #endif
 
-#if defined (WIN32)
-#define strkscmp stricmp
+#if defined(_WIN32) || defined(_WIN64)
+#define strkscmp _stricmp
+#define strdup _strdup
 #else
 #define strkscmp strcasecmp
 #endif
 
 static char *trim(char *str)
 {
-    if(str != NULL)
-    {
-        int len = (int)strlen(str);
-        char *start = str;
-        char *end = str + len;
-        while(start < end && *start <= ' ')
-        {
-            start++;
-        }
-        while(end > start && *(end-1) <= ' ')
-        {
-            end--;
-        }
-        if(end - start != len)
-        {
-            strncpy(str, start, end-start);
-            str[end-start] = '\0';
-        }
-    }
-    return str;
+	if(str != NULL) {
```

Comment:
trim can be moved to utils, or we can introduce a file strutils.c to hold all related functions.